lib/rucio/tests/test_abacus_rse.py

```diff
@@ -0,0 +1,88 @@
+from rucio.daemons import undertaker
+from rucio.daemons.abacus import rse
+from rucio.daemons.judge import cleaner
+from rucio.daemons.reaper import reaper
+from rucio.tests.common import file_generator
+
+
+class TestAbacusRSE():
+    def setUp(self):
+        self.account = 'root'
+        self.scope = 'mock'
+        self.upload_client = UploadClient()
+        self.file_sizes = 2
+        self.rse = 'MOCK4'
+        self.rse_id = get_rse(self.rse).id
+        self.session = get_session()
+
+    def tearDown(self):
+        undertaker.run(once=True)
+        cleaner.run(once=True)
+        reaper.run(once=True, rses=[self.rse], greedy=True)
+
+    def test_abacus_rse(self):
+        """ ABACUS (RSE): Test update of RSE usage. """
+        # Get RSE usage of all sources
+        self.session.query(models.UpdatedRSECounter).delete()
+        self.session.query(models.RSEUsage).delete()
+        self.session.commit()
+
+        # Upload files -> RSE usage should increase
+        self.files = [{'did_scope': self.scope, 'did_name': 'file_' + generate_uuid(), 'path': file_generator(size=self.file_sizes), 'rse': self.rse, 'lifetime': -1} for i in range(0, 2)]
+        self.upload_client.upload(self.files)
+        [os.remove(file['path']) for file in self.files]
+        rse.run(once=True)
+        rse_usage = get_rse_usage(rse=self.rse)[0]
+        assert_equal(rse_usage['used'], len(self.files) * self.file_sizes)
+        rse_usage_from_rucio = get_rse_usage(rse=self.rse, source='rucio')[0]
+        assert_equal(rse_usage_from_rucio['used'], len(self.files) * self.file_sizes)
+        rse_usage_from_unavailable = get_rse_usage(rse=self.rse, source='unavailable')
+        assert_equal(len(rse_usage_from_unavailable), 0)
+
+        # Update RSE usage from unavailable sources
+        update_replica_state(self.rse, self.scope, self.files[0]['did_name'], constants.ReplicaState.UNAVAILABLE)
+        rse.run(once=True)
+        rse_usage = get_rse_usage(rse=self.rse, source='unavailable')[0]
+        assert_equal(rse_usage['used'], self.file_sizes)
+        rse_usage_from_rucio = get_rse_usage(rse=self.rse, source='rucio')[0]
+        assert_equal(rse_usage_from_rucio['used'], len(self.files) * self.file_sizes)
+        rse_usage_from_unavailable = get_rse_usage(rse=self.rse, source='unavailable')[0]
+        assert_equal(rse_usage_from_unavailable['used'], self.file_sizes)
+
+        # Delete files -> rse usage should decrease
+        cleaner.run(once=True)
+        reaper.run(once=True, rses=[self.rse], greedy=True)
+        rse.run(once=True)
+        rse_usage = get_rse_usage(rse=self.rse)[0]
+        assert_equal(rse_usage['used'], 0)
+        rse_usage_from_rucio = get_rse_usage(rse=self.rse, source='rucio')[0]
+        assert_equal(rse_usage_from_rucio['used'], 0)
+        rse_usage_from_unavailable = get_rse_usage(rse=self.rse, source='unavailable')[0]
+        assert_equal(rse_usage_from_unavailable['used'], 2)
```

Comment:
I would expect this to be 0 after the files are deleted, but was not sure about the sources stuff. I guess it is not updating because after the [group by](https://github.com/rucio/rucio/blob/e9b2d634c767d988e8620512fe3c92b04c4c1ab6/lib/rucio/core/rse_counter.py#L162) the list is empty and then the RSEUsage for source=unavailable does not get updated.